experimental/harfbuzz/src/lib.rs

```diff
@@ -0,0 +1,324 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// https://github.com/unicode-org/icu4x/blob/main/docs/process/boilerplate.md#library-annotations
+#![cfg_attr(not(any(test, feature = "std")), no_std)]
+#![cfg_attr(
+    not(test),
+    deny(
+        clippy::indexing_slicing,
+        clippy::unwrap_used,
+        clippy::expect_used,
+        clippy::panic,
+        clippy::exhaustive_structs,
+        clippy::exhaustive_enums,
+        // TODO(#2266): enable missing_debug_implementations,
+    )
+)]
+#![warn(missing_docs)]
+
+//! Using ICU4X as the Unicode Database back end for HarfBuzz.
+
+extern crate alloc;
+mod error;
+
+use crate::error::HarfBuzzError;
+use alloc::boxed::Box;
+use core::ffi::c_void;
+use harfbuzz_sys::*;
+use icu_normalizer::properties::CanonicalCombiningClassMap;
+use icu_normalizer::properties::CanonicalComposition;
+use icu_normalizer::properties::CanonicalDecomposition;
+use icu_normalizer::properties::Decomposed;
+use icu_normalizer::provider::CanonicalCompositionsV1Marker;
+use icu_normalizer::provider::CanonicalDecompositionDataV1Marker;
+use icu_normalizer::provider::CanonicalDecompositionTablesV1Marker;
+use icu_normalizer::provider::NonRecursiveDecompositionSupplementV1Marker;
+use icu_properties::maps::CodePointMapData;
+use icu_properties::provider::GeneralCategoryV1Marker;
+use icu_properties::GeneralCategory;
+use icu_provider::prelude::*;
+
+/// HarfBuzz uses a C enum, which is wider than `u8` but only uses implied
+/// values, so the values stay in the `u8` range. Let's use that to pack this.
+/// Note: ICU4X does not declare the enum items in their numeric order!
+static ICU4X_GENERAL_CATEGORY_TO_HARFBUZZ: [u8; 30] = [
```

Comment:
Nit: The main issue here is that the numerical values that HarfBuzz assigns to the general categories are different than the ones that ICU4C and ICU4X assign. However, this is not the case for CCC, because that is a numeric property defined by Unicode. It would be helpful for the comment to highlight this.

Reply:
Elaborating comment